module/platform/NUsense/HardwareIO/src/HardwareIO.cpp

```diff
@@ -0,0 +1,55 @@
+#include "extension/Configuration.hpp"
+#include "HardwareIO.hpp"
+#include "NUsense/SIProcessor.hpp"
+
+
+namespace module::platform::NUsense {
+
+    using extension::Configuration;
+    using message::platform::RawSensors;// This will most likely change to a more specific message for NUsense
+
+    HardwareIO::HardwareIO(std::unique_ptr<NUClear::Environment> environment) : Reactor(std::move(environment)), nusense() {
+
+        on<Configuration>("HardwareIO.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file HardwareIO.yaml
+            this->log_level = config["log_level"].as<NUClear::LogLevel>();
+            cfg.nusense.port        = config["nusense"]["port"].as<std::string>();
+            cfg.nusense.baud        = config["nusense"]["baud"].as<int>();
+        });
+
+        on<Startup>().then("NUSense HardwareIO Startup", [this] {
+            // Make sure the port is closed before we open it
+            if (nusense.connected()){
+                nusense.close();
+            }
+
+            // If /dev/ttyACM0 is acting up, follow the link below
+            // https://stackoverflow.com/questions/40951728/avrdude-ser-open-cant-open-device-dev-ttyacm0-device-or-resource-busy
```

Comment:
```suggestion
```

Reply:
I'm not sure whether we should remove it. I put the link there because it was a fix for nix based machines not seeing the port when we connect the USB to it